wgpu-hal/src/metal/mod.rs

```diff
@@ -356,6 +356,10 @@ impl Queue {
             timestamp_period,
         }
     }
+
+    pub fn as_raw(&self) -> parking_lot::MutexGuard<'_, metal::CommandQueue> {
+        self.raw.lock()
+    }
```

Comment:
Hmmmm. I think either you should hand back a reference to the lock itself, return `impl Deref<Target = metal::CommandQueue>`. I think the first is probably best, as it allows the user to control the locking